### DISCUSSION TITLE
How to make a translation into Russian?

### DISCUSSION THREAD

**Discussion by Hkr32** [vuepress-theme-hope/vuepress-theme-hope#1705] [Q&A] [answered]
I added
![image](https://user-images.githubusercontent.com/12296392/164971081-2888af47-d6c8-4cbc-b34c-6636b3b47c15.png)
and got an error
![image](https://user-images.githubusercontent.com/12296392/164971120-f1966486-040f-4fb4-9238-007f891aa00a.png)

```yarn docs:vite-serve
yarn run v1.22.1
$ vuepress-vite dev src
info Initializing VuePress and preparing data...
error error in hook extendsPage from vuepress-plugin-copyright2
TypeError: Cannot read property 'replace' of undefined
    at Object.extendsPage [as hook] (C:\projects\vue\vuepress-theme-hope\packages\copyright2\lib\node\plugins.js:22:33)
    at Object.process (C:\projects\vue\vuepress-theme-hope\node_modules\@vuepress\core\lib\pluginApi\createHookQueue.js:26:48)
    at async createPage (C:\projects\vue\vuepress-theme-hope\node_modules\@vuepress\core\lib\page\createPage.js:132:5)
    at async Promise.all (index 19)
    at async resolveAppPages (C:\projects\vue\vuepress-theme-hope\node_modules\@vuepress\core\lib\app\resolveAppPages.js:18:19)
    at async appInit (C:\projects\vue\vuepress-theme-hope\node_modules\@vuepress\core\lib\app\appInit.js:22:17)
    at async dev (C:\projects\vue\vuepress-theme-hope\node_modules\@vuepress\cli\lib\commands\dev\createDev.js:48:9)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by Mister-Hope** [comment]
You can contribute to https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/main/packages/copyright2/src/node/locales.ts to let that plugin support `ru`.

Otherwise you need to set [locales](https://vuepress-theme-hope.github.io/v2/copyright/config.html#locales) for ru in `themeConfig.plugins.copyright`

**Reply by Mister-Hope** [reply]
Also we sincerely welcome you to check & improve exising locales and add missing locales for all plugins and themes.

They should be all available under `packages/<plugin-or-theme-name>/src/node/locale.ts`

**Reply by Hkr32** [reply]
@Mister-Hope, I can try. Is it possible to do this with different PRs or can it be done with one?

**Reply by Mister-Hope** [reply]
Both of them are  fine

**Comment by Mister-Hope** [comment, marked as answer]
These kind of issues can be avoided, so I preffer this is an issue in `getLocales` helper,  so I made an improvement in https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/3b7b3a20674c1ad9e8bcb9bcb991a2b0ea7bf9ee. Now unknown locale will gives warning and fallback to root locale (if root locale can be recognized) or `en-US`(root locale cannot be recognized).

**Comment by Hkr32** [comment]
If I'm in the root directory, how do I start theme documentation development?
`yarn run docs/theme:serve` not found
Or is it possible only if you enter the desired folder and run from under it?

**Reply by Mister-Hope** [reply]
Could you decribe more details what you are asking? You already run that and provide logs in the above dicussion.

**Reply by Hkr32** [reply]
![image](https://user-images.githubusercontent.com/12296392/164976493-37c2ee91-cdf8-4796-b555-92928fa9ce1e.png)
...
![image](https://user-images.githubusercontent.com/12296392/164976502-ddf9c109-5723-4991-9497-d4c3d840f2d0.png)

**Reply by Hkr32** [reply]
https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/427bc4754a987d926c01274be5e8a32be6ff4688/docs/theme/src/contribution.md?plain=1#L98

**Reply by Mister-Hope** [reply]
```bash
yarn workspace xxx-docs docs:vite-serve
# or
yarn workspace xxx-docs docs:webpack-serve
```

Fot theme docs:

```bash
yarn workspace theme-docs docs:webpack-serve
```

**Reply by Mister-Hope** [reply]
I will update that, thanks for reminding this

**Reply by Mister-Hope** [reply]
Updated in https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/c03239f1e5c764f12ce862ebaddc7d6021f37607

**Comment by Hkr32** [comment]
@Mister-Hope, i started translating theme into Russian. Make a PR now or when everything is ready?

**Reply by Mister-Hope** [reply]
I already merged your pr and published a new version

**Reply by Hkr32** [reply]
I'm talking about the documentation for theme)
![image](https://user-images.githubusercontent.com/12296392/165138425-8647bc73-857d-4a64-a5a5-c5829e6490a2.png)

**Reply by Mister-Hope** [reply]
Oh sorry for the late reply, welcome to do it after we merge #1731.

(BTW, we are making a huge refactor in our docs)
